Library/Homebrew/extend/os/mac/unpack_strategy/zip.rb

```diff
@@ -2,26 +2,39 @@ module UnpackStrategy
   class Zip
     prepend Module.new {
       def extract_to_dir(unpack_dir, basename:, verbose:)
-        volumes = super.stderr.chomp
-                       .split("\n")
-                       .map { |l| l[/\A   skipping: (.+)  volume label\Z/, 1] }
-                       .compact
+        result = super
 
-        return if volumes.empty?
+        volumes = result.stderr.chomp
+                        .split("\n")
+                        .map { |l| l[/\A   skipping: (.+)  volume label\Z/, 1] }
+                        .compact
 
-        Dir.mktmpdir do |tmp_unpack_dir|
-          tmp_unpack_dir = Pathname(tmp_unpack_dir)
+        unless volumes.empty?
+          Dir.mktmpdir do |tmp_unpack_dir|
+            tmp_unpack_dir = Pathname(tmp_unpack_dir)
 
-          # `ditto` keeps Finder attributes intact and does not skip volume labels
-          # like `unzip` does, which can prevent disk images from being unzipped.
-          system_command! "ditto",
-                          args: ["-x", "-k", path, tmp_unpack_dir],
-                          verbose: verbose
+            # `ditto` keeps Finder attributes intact and does not skip volume labels
+            # like `unzip` does, which can prevent disk images from being unzipped.
+            system_command! "ditto",
+                            args: ["-x", "-k", path, tmp_unpack_dir],
+                            verbose: verbose
 
-          volumes.each do |volume|
-            FileUtils.mv tmp_unpack_dir/volume, unpack_dir/volume, verbose: verbose
+            volumes.each do |volume|
+              FileUtils.mv tmp_unpack_dir/volume, unpack_dir/volume, verbose: verbose
+            end
           end
         end
+
+        return if result.stderr.lines.grep(/^\._/).empty?
+
+        system_command!("dot_clean",
+                        args: [
+                          "-m#{"v" if verbose}",
```

Comment:
Actually, I think this can stay, since we don't print to stdout anyways.

Reply:
Actually, `dot_clean` prints to stderr, which is hidden thanks to `print_stderr: false`.